ckanext/xloader/loader.py

```diff
@@ -159,6 +159,13 @@ def load_csv(csv_filepath, resource_id, mimetype='text/csv', logger=None):
 
     # TODO worry about csv header name problems
     # e.g. duplicate names
+    # (canada fork only): not worrrying about dupes as that is handled in ckanext-validation
+
+    # (canada fork only): remove underscores from front of headers as it is not supported in datastore
+    #                     see: ckanext.datastore.helpers.is_valid_field_name
+    for i, header in enumerate(headers):
+        if header.startswith('_'):
+            headers[i] = header.lstrip('_')
```

Comment:
If the goal is to get the data loaded however we can, including changing column names, we need a function to pass all the column names and that function would be responsible to munge them and generate unique names that follow all the rules (underscores, whitespace etc)

If I was writing such a function I would *add* a prefix for columns starting with an underscore instead of stripping them, like `'_id'` becomes `'x_id'` or something. To make them unique we need to add an in incrementing number to names that conflict, but also have to keep the name under the maximum length.

This could be a helper function in the extension. Even if it's not used in the templates a helper can be overridden by extensions easily if desired.